auth/bearer/token.go

```diff
@@ -0,0 +1,49 @@
+package bearer
+
+import (
+	"context"
+	"time"
+)
+
+// Token provides a type wrapping a bearer token and expiration metadata.
+type Token struct {
+	Value string
+
+	CanExpire bool
+	Expires   time.Time
+}
+
+// IsExpired returns if the token's Expires time is before or equal to the time
+// provided. If CanExpires is false, Expired will always return false.
+func (t Token) Expired(now time.Time) bool {
+	if !t.CanExpire {
+		return false
+	}
+	return now.Round(0).Equal(t.Expires) || now.After(t.Expires)
```

Comment:
Should we do `now = now.Round(0)` before the return? That way `Equal` and `After` are both wall-clock checks instead of monotonic.